### DISCUSSION TITLE
log4j upgrade

### DISCUSSION THREAD

**Discussion by michaelwechner** [wyona/yanel#92] [Show and tell]
Hi Together

Please do the following steps.

Upgrade to log4j 2.17.0
---------------------

git diff src/build/dependencies.xml
diff --git a/src/build/dependencies.xml b/src/build/dependencies.xml
index f6b9640..8335316 100644
--- a/src/build/dependencies.xml
+++ b/src/build/dependencies.xml
@@ -54,8 +54,8 @@
       <dependency groupId="org.bouncycastle" artifactId="bcpg-jdk16" version="1.46"/>

       
-      <dependency groupId="org.apache.logging.log4j" artifactId="log4j-api" version="2.0-beta8"/>
-      <dependency groupId="org.apache.logging.log4j" artifactId="log4j-core" version="2.0-beta8"/>
+      <dependency groupId="org.apache.logging.log4j" artifactId="log4j-api" version="2.17.0"/>
+      <dependency groupId="org.apache.logging.log4j" artifactId="log4j-core" version="2.17.0"/>

       
       <dependency groupId="log4j" artifactId="log4j" version="1.2.16"/>

Get log4j 2.17.0 from https://repo1.maven.org/maven2/
-------------------------------------------------

git diff src/build/build.properties
diff --git a/src/build/build.properties b/src/build/build.properties
index 5a21348..6b6103e 100644
--- a/src/build/build.properties
+++ b/src/build/build.properties
@@ -153,7 +153,8 @@ local.config.dir=conf/local
 #local.config.home=/path/to/project/yanel-configuration

 # Wyona Maven Repo URL
-maven.url=http://maven2.wyona.org/
+#maven.url=http://maven2.wyona.org/
+maven.url=https://repo1.maven.org/maven2/

 # Tomcat version shipped with Yanel, whereas it will be downloaded from http://maven2.wyona.org/apache-org/apache-tomcat/
 tomcat.version=7.0.25

Update ThreadContext
--------------------

git diff src/webapp/src/java/org/wyona/yanel/servlet/YanelServlet.java
diff --git a/src/webapp/src/java/org/wyona/yanel/servlet/YanelServlet.java b/src/webapp/src/java/org/wyona/yanel/servlet/YanelServlet.java
index 2f1bac521..92ce4fa41 100644
--- a/src/webapp/src/java/org/wyona/yanel/servlet/YanelServlet.java
+++ b/src/webapp/src/java/org/wyona/yanel/servlet/YanelServlet.java
@@ -437,7 +437,7 @@ public class YanelServlet extends HttpServlet {
             log.error(e, e);
             throw new IOException(e.getMessage());
         } finally {
-            ThreadContext.clear();
+            ThreadContext.clearAll();
         } // NOTE: This was our last chance to log an exception, hence do not add code outside the try-catch block
     }

Remove old libs
--------------

rm local/apache-tomcat-7.0.25/webapps/yanel/WEB-INF/lib/log4j-api-2.0-beta8.jar
rm local/apache-tomcat-7.0.25/webapps/yanel/WEB-INF/lib/log4j-core-2.0-beta8.jar

Restart Yanel

Hope that helps

Michael

**Comment by michaelwechner** [comment]
you should also do the  following change

git diff src/build/pom-core.xml
diff --git a/src/build/pom-core.xml b/src/build/pom-core.xml
index aa6bb3786..4d61fc8f3 100644
--- a/src/build/pom-core.xml
+++ b/src/build/pom-core.xml
@@ -64,7 +64,7 @@
     <dependency>
       <groupId>org.apache.logging.log4j</groupId>
       <artifactId>log4j-api</artifactId>
-      <version>2.0-beta8</version>
+      <version>2.17.0</version>
     </dependency>

whereas I have committed the changes now

Thanks

Michael

**Comment by baszero** [comment]
I think with the current version of Yanel you also have to comment out the following resources in `conf/resource-types.xml`:

```
  <resource-type src="@YANEL_SRC_DIR@/src/resources/odt/" compile="true" copy-dir-name="yanel-odt"/> 
  <resource-type src="@YANEL_SRC_DIR@/src/resources/user-mgmt/" compile="true" copy-dir-name="yanel-user-mgmt"/>
  <resource-type src="@YANEL_SRC_DIR@/src/resources/xinha/" compile="true" copy-dir-name="yanel-xinha"/>
```

Otherwise you get the old log4j version in the final WEB-INF/lib.
